### ISSUE TITLE
GitHub Issue + Milestones

### ISSUE THREAD

**Issue by janzeteachesit** [janzeteachesit/Learning-Diary#17]
- [ ] [Mastering Issues](https://guides.github.com/features/issues/)

* A title and description describe what the issue is all about. 
* Color-coded labels help you categorize and filter your issues (just like labels in email).
* One assignee is responsible for working on the issue at any given time.
* Comments allow anyone with access to the repository to provide feedback.

* A milestone acts like a container for issues. This is useful for associating issues with specific features or project phases (e.g. Weekly Sprint 9/5-9/16 or Shipping 1.0).  Milestones are groups of issues that correspond to a project, feature, or time period. People use them in many different ways in software development. Some examples of milestones on GitHub include:
  * Beta Launch — File bugs that you need to fix before you can launch the beta of your project. It’s a great way to make sure you aren’t missing anything.
  * October Sprint — File issues that you’d like to work on in October. A great way to focus your efforts when there’s a lot to do.
  * Redesign — File issues related to redesigning your project. A great way to collect ideas on what to work on.

**Comment by janzeteachesit** [comment]
- [ ] [How to manage github issues for (priority, etc)? [closed]](https://softwareengineering.stackexchange.com/questions/129714/how-to-manage-github-issues-for-priority-etc)

**Comment by janzeteachesit** [comment]
- [ ] [How we organize GitHub issues: A simple styleguide for tagging](https://robinpowered.com/blog/best-practice-system-for-organizing-and-tagging-github-issues/)

**Comment by janzeteachesit** [comment]
- [ ] [GITHUB LABELS AND MILESTONES](https://docs.saltstack.com/en/latest/topics/development/labels.html)

* All issues are assigned to a milestone, whereas pull requests are almost never assigned to a milestone as the mean lifetime of pull requests is short enough that there is no need to track them temporally.
* uses milestones to indicate which issues are blocked on submitter or upstream actions, are approved, or are scheduled to be fixed or implemented in an upcoming sprint. If an issue is not attached to a sprint milestone, you are welcome to work on it at your own desire and convenience. If it is attached to a sprint milestone and you have already begun working on it or have a solution in mind or have other ideas related to the issue, you are encouraged to coordinate with the assignee via the GitHub issue tracker to create the best possible solution or implementation.
  * Approved - The issue has been validated and has all necessary information.
  * Blocked - The issue is waiting on actions by parties outside of SaltStack, such as receiving more information from the submitter or resolution of an upstream issue. This milestone is usually applied in conjunction with the labels Info Needed, Question, Expected Behavior, Won't Fix For Now, or Upstream Bug.
  * Under Review - The issue is having further validation done by a SaltStack engineer.
  * <Sprint> - The issue is being actively worked on. Sprint milestones names are constructed from the chemical symbol of the next release's codename and the number of sprints until that release is made. For example, if the next release codename is Neon and there are five sprints until that release, the corresponding sprint milestone will be called Ne 5. See <version-numbers> for a discussion of Salt's release codenames.
